core/src/main/scala/org/apache/spark/shuffle/ShuffleChecksumUtils.scala

```diff
@@ -22,9 +22,22 @@ import java.util.zip.CheckedInputStream
 
 import org.apache.spark.network.shuffle.checksum.ShuffleChecksumHelper
 import org.apache.spark.network.util.LimitedInputStream
+import org.apache.spark.shuffle.IndexShuffleBlockResolver.NOOP_REDUCE_ID
+import org.apache.spark.storage.{BlockId, ShuffleChecksumBlockId, ShuffleDataBlockId}
 
 object ShuffleChecksumUtils {
 
+  /**
+   * Return checksumFile for shuffle data block ID. Otherwise, null.
+   */
+  def getChecksumFileName(blockId: BlockId, algorithm: String): String = blockId match {
+    case ShuffleDataBlockId(shuffleId, mapId, _) =>
+      ShuffleChecksumHelper.getChecksumFileName(
+        ShuffleChecksumBlockId(shuffleId, mapId, NOOP_REDUCE_ID).name, algorithm)
```

Comment:
Isn't `NOOP_REDUCE_ID` for sort-based shuffle only?

Reply:
It's the only `BlockManager` implementation in order to get checksum file for now. Here.

https://github.com/apache/spark/blob/47d801e5e9ded3fb50d274a720ee7874e0b37cc3/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L298-L307

https://github.com/apache/spark/blob/47d801e5e9ded3fb50d274a720ee7874e0b37cc3/core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala#L551-L557

Reply:
`reducerID` is ignored.

Reply:
Yea, looks so.